core/dvector.h

```diff
@@ -466,6 +467,16 @@ class PoolVector {
 		return OK;
 	}
 
+	String join(String delimiter) {
+		String rs = "";
+		int s = size();
+		Read r = read();
+		for(int i=0;i<s;i++) {
+			rs += r[i] + delimiter;
+		}
+		rs.erase( rs.length()-1, 1);
```

Comment:
You have an issue here, what if the delimiter is more than one character long? Wouldn't part of it linger around then?